src/Algebra/Definitions.agda

```diff
@@ -26,21 +25,122 @@ module Algebra.Definitions
 open import Algebra.Core using (Op₁; Op₂)
 open import Data.Product.Base using (_×_; ∃-syntax)
 open import Data.Sum.Base using (_⊎_)
+open import Function.Base using (flip; id; _∘_; _on_)
+open import Relation.Nullary.Negation.Core using (¬_)
+open import Relation.Unary using (Pred)
+
+------------------------------------------------------------------------
+-- Properties of operations
+
+------------------------------------------------------------------------
+-- A generalisation: because not everything can be defined in terms of
+-- a single relation _≈_, AND because the corresponding definitions in
+-- `Relation.Binary.Core` use *implicit* quantification... so there is
+-- plenty of potential for later/downstream refactoring of this design
+
+module Monotonicity {ℓ₁ ℓ₂} (_≤₁_ : Rel A ℓ₁) (_≤₂_ : Rel A ℓ₂) where
+
+  Monotone₁ : Op₁ A → Set _
+  Monotone₁ f = ∀ x y → x ≤₁ y → (_≤₂_ on f) x y
+
+  Monotone₂ : Op₂ A → Set _
+  Monotone₂ _∙_ = ∀ x y u v → x ≤₁ y → u ≤₁ v → (x ∙ u) ≤₂ (y ∙ v)
```

Comment:
This looks very similar to `Congruent₂` to me. I wonder if there's something there.

Reply:
Well, you're correct, of course, but this is precisely because the `Congruent*`/`Relation.Binary.Preserves*` properties use *implicit* quantification (because the relation instances can be inferred by the typechecker), distinct from the design decision made in issue #1436 to make the `Cancellative` properties take *explicit* arguments. I chose `Monotone` etc. because at one stage I was trying to have a single design in which both the `Monotone` and `Cancellative` deifnitions could be derived from a single family. But it required a fair amount of `_on_` gymnastics, plus redundant uses of `id`, and I decided to inline everything by hand. 

But the general point: we almost never use `Congruent*`, but definitely *do* use `Monotone` a lot, still stands. 

So I would pleased if this specimen design prompts a wider rethink. There are a number of outstanding issues even in the v2.0 milestone which I think could be tackled with a design along these lines, if we could only think about it enough and agree... so this contribution is offered up towards such an end.